client/cmd/translationsreport/main.go

```diff
@@ -73,3 +85,125 @@ type WorksheetEntry struct {
+			entryErrs = append(entryErrs, fmt.Sprintf("no context found for report with ID %q", e.ID))
+			continue
+		}
+		sortedEntries[e.Context] = append(sortedEntries[e.Context], e)
+	}
+
+	entryErrs = append(entryErrs, printCoreLines(sortedEntries["notifications"])...)
+	entryErrs = append(entryErrs, printJSLines(sortedEntries["js"])...)
+	entryErrs = append(entryErrs, printHTMLLines(sortedEntries["html"])...)
+
+	if len(entryErrs) == 0 {
+		return nil
+	}
+
+	fmt.Printf("\nErrors encountered\n")
+	for i := range entryErrs {
+		fmt.Println(entryErrs[i])
+	}
+
+	return errors.New("completed with errors")
+}
+
+func printCoreLines(es []*WorksheetEntry) (entryErrs []string) {
+	fmt.Printf("\nCore notifications. Put in correct locale map in client/core/locale_ntfn.go\n------------\n")
+	noteEntries := make(map[string][2]*WorksheetEntry)
+	for _, e := range es {
+		var i int
+		var entryID string
+		switch {
+		case strings.HasSuffix(e.ID, " subject"):
+			entryID = e.ID[:len(e.ID)-len(" subject")]
+		case strings.HasSuffix(e.ID, " template"):
+			i = 1
+			entryID = e.ID[:len(e.ID)-len(" template")]
+		default:
+			entryErrs = append(entryErrs, fmt.Sprintf("corrupted ID %q for web notification", e.ID))
+			continue
+		}
+		ne := noteEntries[entryID]
+		ne[i] = e
+		noteEntries[entryID] = ne
+	}
+
+	var unpaired []*WorksheetEntry
+	for entryID, st := range noteEntries {
+		subjectEntry, templateEntry := st[0], st[1]
+		if subjectEntry == nil {
+			if templateEntry == nil {
+				entryErrs = append(entryErrs, fmt.Sprintf("Missing subject or template for translation with ID %s", entryID))
+			} else {
+				unpaired = append(unpaired, templateEntry)
+			}
+			continue
+		} else if templateEntry == nil {
+			unpaired = append(unpaired, subjectEntry)
+			continue
+		}
+		fmt.Printf(webNoteTemplate, entryID, formatReportWithType(subjectEntry), formatReportWithType(templateEntry))
+	}
+	for _, e := range unpaired {
+		fmt.Println("Topic"+e.ID, ": ", formatReportWithType(e)+",")
+	}
+
+	return
+}
+
+func printJSLines(es []*WorksheetEntry) (entryErrs []string) {
+	fmt.Printf("\nJS Notifications. Add these to the appropriate map in client/webserver/jsintl.go, replacing the IDs with the appropriate variables\n------------\n")
+	for _, e := range es {
+		fmt.Printf("	%s: %s,\n", e.ID, formatReportEntry(e))
+	}
+	return
+}
+
+func printHTMLLines(es []*WorksheetEntry) (entryErrs []string) {
+	fmt.Printf("\nHTML Notifications. Add these to the language file in client/webserver/locales\n------------\n")
+	for _, e := range es {
+		fmt.Printf(`	"%s": %s,`+"\n", e.ID, formatReportEntry(e))
+	}
+	return
+}
+
+func escapeTemplate(s string) string {
+	b, _ := json.Marshal(s)
+	return string(b)
+}
+
+func formatReportEntry(e *WorksheetEntry) string {
+	var v, n string
+	if e.Version > 0 {
+		v = fmt.Sprintf("Version: %d, ", e.Version)
+	}
+	t := escapeTemplate(e.Translation)
+	if e.Notes != "" {
+		n = fmt.Sprintf(", Notes: %s", escapeTemplate(e.Notes))
+	}
```

Comment:
I don't actually need these notes in the translations I guess, just in the English. Will probably remove.